jot/JotTextEditView.m

```diff
@@ -85,10 +85,14 @@ - (instancetype)init
             UIView *buttonView = [[UIView alloc] initWithFrame:CGRectMake(0,0,40,40)];
             [buttonView addSubview:colorButton];
             [colorSelectorItems addObject:[[UIBarButtonItem alloc] initWithCustomView:buttonView]];
+
+            // Add a UIBarButtonItem with FlexibleSpace to evenly distribute BarButtons across UIToolbar
+            [colorSelectorItems addObject:[[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:nil action:nil]];
```

Comment:
Can you explain this fix a little? As I understand it, we're adding another `UIBarButtonItem` that enables flexible spacing...? How does that work?

Reply:
It helps distribute our `UIBarButtonItems` evenly because we can't use AutoLayout. Otherwise UIToolbar, given an array of `UIBarButtonItems` will first fill the left most spot in the `UIToolbar` and then all additional buttons and added from the right side. This caused a 
UI quirk (that was most glaring on larger screens) where there was an awkward visual gap between the two left-most buttons:

![img_0566](https://user-images.githubusercontent.com/465289/33849161-a8c5bdfc-de64-11e7-833b-731b40e9c708.jpg)

It's definitely easier to explain visually - here's how it works by default. A single `UIBarButtonItem` in a `UIToolbar`

1 _ _ _ _ _ _ _ _ _

Eg) Two `UIBarButtonItems` in a `UIToolbar`

1 _ _ _ _ _ _ _ _ 2

Eg) Three `UIBarButtonItems`

1 _ _ _ _ _ _ _ 2 3

In our situations we want the color picker `UIBarButtonItems` evenly distributed so we use this [kinda special UIBarButton that gives us a:](https://developer.apple.com/documentation/uikit/uibarbuttonsystemitem/uibarbuttonsystemitemflexiblespace?language=objc)
> Blank space to add between other items. The space is distributed equally between the other items

which effectively gives us what we want without AutoLayout. With added flexible space `UIBarButtonItems`:

![img_0567](https://user-images.githubusercontent.com/465289/33849186-bea8a9ea-de64-11e7-9830-79cfeea6d44d.jpg)

Reply:
Woaaah. Thank you for the thorough explanation! This is really helpful and I understand it now.